actionview/lib/action_view/helpers/url_helper.rb

```diff
@@ -664,7 +664,7 @@ def to_form_params(attribute, namespace = nil)
             params << { name: namespace, value: attribute.to_param }
           end
 
-          params.sort_by { |pair| pair[:name] }
+          params.sort_by { |pair| pair[:name].to_s }
```

Comment:
I think it would make more sense to cast to a string a few lines before in the `case`'s `else` block: `params << { name: namespace.to_s, value: attribute.to_param }`

Reply:
Agree